src/interface/core/lang_config.ts

```diff
@@ -11,29 +11,73 @@ if (Languages[StaticConfig.data.appLanguage]) {
 	const data = Languages[StaticConfig.data.appLanguage].translations
 	initialTranslations = data
 } else {
-	initialTranslations = Languages.en.translations
+	initialTranslations = Languages.english.translations
 	const err = `Couldnt find language by name ${StaticConfig.data.appLanguage}`
 	throwError(err, err)
 }
 
 const LanguageState: PuffinState = new state({
 	translations: initialTranslations,
-	fallbackTranslations: Languages.en.translations,
+	fallbackTranslations: Languages.english.translations,
 })
 
 function setFallback(notFoundLang: string): void {
-	StaticConfig.data.appLanguage = 'en'
+	StaticConfig.data.appLanguage = 'english'
 	LanguageState.data.translations = Languages[StaticConfig.data.appLanguage].translations
 	const err = `Couldnt find language by name ${notFoundLang}`
 	throwError(err, err)
 }
 
 StaticConfig.keyChanged('appLanguage', (newLanguage: string) => {
 	if (Languages[newLanguage].name === 'Default') {
-		if (Languages[globalAny.navigator.language]) {
-			LanguageState.data.translations = Languages[globalAny.navigator.language].translations
-		} else {
-			setFallback(globalAny.navigator.language)
+		// Get the language part from the locale (e.g. en-GB -> en)
+		const locale = globalAny.navigator.language.substr(0, globalAny.navigator.language.indexOf('-'))
+
+		switch (locale) {
```

Comment:
I don't really think this switch is needed, each language already provides  a locale, example: https://github.com/Graviton-Code-Editor/Graviton-App/blob/09618d059bcd358171bb4015cc5c4a5a12c836df/languages/english.json#L3